addons/analytic/static/src/js/analytic_distribution/analytic_distribution.js

```diff
@@ -0,0 +1,606 @@
+            onClick: (ev) => this.tagClicked(ev, dist_tag.id),
+            onDelete: this.editingRecord ? () => this.deleteTag(dist_tag.id) : undefined
+        }));
+    }
+
+    get listForJson() {
+        let res = {};
+        this.listReady.map(({analytic_account_id, percentage}) => {
+            res[parseInt(analytic_account_id)] = percentage;
+        });
+        return res;
+    }
+
+    get firstIncompletePlanId() {
+        for (let group_id in this.list) {
+            let group_status = this.groupStatus(group_id);
+            if (["orange", "red"].includes(group_status)) return group_id;
+        }
+        return 0;
+    }
+
+    get existingAnalyticAccountIDs() {
+        return this.listFlat.filter((i) => !!i.analytic_account_id).map((i) => i.analytic_account_id);
+    }
+
+    get listReady() {
+        return this.listFlat.filter((dist_tag) => this.tagIsReady(dist_tag));
+    }
+
+    get listFlat() {
+        return Object.values(this.list).flatMap((g) => g.distribution);
+    }
+
+    get list() {
+        return this.state.list;
+    }
+
+    get sortedList() {
+        return Object.values(this.list).sort((a, b) => {
+            const aApp = a.applicability,
+                  bApp = b.applicability;
+            return aApp > bApp ? 1 : aApp < bApp ? -1 : 0;
+        });
+    }
+
+    get allowSave() {
+        for (let group_id in this.list) {
+            if (['orange', 'red'].includes(this.groupStatus(group_id))) return false;
+        }
+        return true;
+    }
+
+    get editingRecord() {
+        return !this.props.readonly;
+    }
+
+    get isDropdownOpen() {
+        return this.state.showDropdown && !!this.dropdownRef.el;
+    }
+
+    applicabilityStatus(group_id) {
+        let group = this.list[group_id];
+        let status = this.groupStatus(group_id);
+        let description = group.applicability.charAt(0).toUpperCase() + group.applicability.slice(1) + " - ";
+        switch(status){
+            case "gray":
+                description += this.env._t("Editing (OK)");
+                break;
+            case "orange": {
+                description += this.env._t("Editing (Incomplete)");
+                break;
+            }
+            case "red": {
+                description += this.env._t("Invalid");
+                break;
+            }
+            case "green": {
+                description += this.env._t("OK");
+                break;
+            }
+        }
+        return description;
+    }
+
+    groupStatus(id) {
+        let group = this.list[id];
+        let ready_tags = this.listReadyByGroup(id);
+        if (group.distribution.length > ready_tags.length) {
+            if (group.applicability === 'mandatory') return 'orange'
+            return 'gray';
+        }
+        let sum = this.sumByGroup(id);
+        if (group.applicability === 'mandatory' && (sum < 99.99 || sum >= 100.01)){
+            return 'red';
+        }
+        return 'green';
+    }
```

Comment:
The statuses should be more explicit in the technical string.
For instance:
* `green` -> `ok`
* `red` -> `nok`
* `gray` -> `edit_ok`
* `orange` -> `edit_nok`